### PR TITLE
Add UC23  Get total correct answers per team

### PR DESCRIPTION
**Pull request by henrymcarvajal**

### CODE REVIEW

File: app/src/service/stats/totalCorrectAnswersPerTeam.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    const scores = [];
+    for (let teamsStat of teamsStats) {
+        const score = {
+            "teamId": teamsStat.team_id,
+            "score": teamsStat.team_score + "/" + (parseInt(teamsStat.member_count) * questionsCount)
+        }
+        scores.push(score);
+    }
+    return scores;
```
Comment (andymotagl) [inline app/src/service/stats/totalCorrectAnswersPerTeam.js:22]: What about:
```javascript
return teamsStats
    .map(ts => {
       teamId: ts.team_id,
       score: `${ts.team_score} / ${(parseInt(ts.member_count) * questionsCount)}`
     });
```

File: app/src/service/stats/statsMan.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const { getTotalGameTimePerTeam } = require('./totalGameTimePerTeam');
+const { getMaxStreakDaysPerTeam } = require('./maxStreakDaysPerTeam');
+const { getTotalGameDaysPerTeam } = require('./totalGameDaysPerTeam');
+const { getTotalCorrectAnswersPerTeam } = require('./totalCorrectAnswersPerTeam');
+
+const AVAILABLE_STATISTICS = {
+    'totalGameTimePerTeam': getTotalGameTimePerTeam,
+    'streakDaysPerTeam': getMaxStreakDaysPerTeam,
+    'totalGameDaysPerTeam': getTotalGameDaysPerTeam,
+    'totalCorrectAnswersPerTeam': getTotalCorrectAnswersPerTeam
```
Comment (andymotagl) [inline app/src/service/stats/statsMan.js:10]: Any  reason to have the keys with literals?

File: app/src/db/stats/totalCorrectAnswersPerTeam.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const GET_TEAM_SCORE_AND_MEMBER_COUNT =
+    `SELECT t.team_id, COUNT(*) member_count, SUM(g.score) team_score        
+       FROM game g
+       JOIN experience e ON e.experience_id = g.experience_id 
+       JOIN player p on p.player_id = g.player_id
+       JOIN team t on t.team_id = p.team_id
```
Comment (andymotagl) [inline app/src/db/stats/totalCorrectAnswersPerTeam.js:6]: Some on to uppercase

File: app/src/service/stats/totalCorrectAnswersPerTeam.js (hunk 1 above)
Comment (henrymcarvajal) [reply app/src/service/stats/totalCorrectAnswersPerTeam.js:22]: Of course, my horse!

File: app/src/db/experience.js [hunk 4]
```diff
@@ -38,10 +38,21 @@ const setQuestions = async (experienceId, questions) => {
   return res.rowCount === 1 ? res.rows[0] : null;
 }
 
+const getQuestionsCount = async (experienceId) => {
+  const selectExperienceQuestionCount =
+      `SELECT count(*) as total
+         FROM experience e, jsonb_array_elements(questions['data']) WITH ORDINALITY record(question, position)
+        WHERE e.experience_id = $1`;
```
[the remark this answers is not in the document]
Comment (henrymcarvajal) [reply app/src/db/experience.js:45]: Nice, thank you!

File: app/src/service/stats/statsMan.js (hunk 2 above)
Comment (henrymcarvajal) [reply app/src/service/stats/statsMan.js:10]: Fixed!

File: app/src/db/stats/totalCorrectAnswersPerTeam.js (hunk 3 above)
Comment (henrymcarvajal) [reply app/src/db/stats/totalCorrectAnswersPerTeam.js:6]: sry, what do you mean?

Comment (FernandoArevaloE) [reply app/src/db/stats/totalCorrectAnswersPerTeam.js:6]: changing  `JOIN player p on p.player_id = g.player_id JOIN team t on t.team_id = p.team_id` to `JOIN player p ON p.player_id = g.player_id JOIN team t ON t.team_id = p.team_id`

Comment (henrymcarvajal) [reply app/src/db/stats/totalCorrectAnswersPerTeam.js:6]: Gotcha! Thanks!

**Review by andymotagl — APPROVED** [review]

**Review by FernandoArevaloE — APPROVED** [review]

**Pull request merged**